lib/lint-file-names.js

```diff
@@ -0,0 +1,44 @@
+const execa = require('execa');
+const path = require('path');
+
+function getProjectRoot() {
+  let cmd = 'git rev-parse --show-toplevel';
+  return execa.command(cmd).then(({ stdout }) => {
+    return stdout.toString().trim();
+  });
+}
+
+function absoluteToRelative(root, absPath) {
+  if (!absPath.startsWith('/')) {
+    throw new Error(`Unexpected non-absolute path: ${absPath}`);
+  }
+  return path.relative(root, absPath);
+}
+
+const dasherizedNamesOnly = /^(@?[a-z0-9][a-zA-Z0-9-]+\/)*(_?[a-z0-9-.][a-zA-Z0-9-.]+)?$/;
+function lintFileName(relativePath) {
+  return dasherizedNamesOnly.test(relativePath);
+}
+
+function lintFileNames(paths) {
+  return getProjectRoot().then(root => {
```

Comment:
Any reason not to use `async`/`await` here like in another file?

Reply:
async/await would be better, yes. But the eslintrc is old and this file doesn't parse.
The only reason I was able to use async/await in in the `bin/*` files is that they aren't linted :)

I'm on the fence about whether it's worth trying to clean this codebase up in a larger sense (revisiting and updating the linting, e.g.), or if it's worth restarting it. There's no good reason that this should be an ember-addon, and we pay a roughly 7x performance penalty for it. Imo it might be better as a smaller, tested, micro linting library.